radis/core/tests/test_processors.py

```diff
@@ -0,0 +1,274 @@
+    job = ExtractionJobFactory.create(owner=user)
+    task = ExtractionTaskFactory.create(job=job)
+
+    processor = AnalysisTaskProcessor(task)
+
+    assert processor.task == task
+
+
+@pytest.mark.django_db
+def test_start_with_canceled_job_status_canceling():
+    user = UserFactory.create()
+    job = ExtractionJobFactory.create(owner=user, status=AnalysisJob.Status.CANCELING)
+    task = ExtractionTaskFactory.create(job=job, status=AnalysisTask.Status.PENDING)
+
+    processor = AnalysisTaskProcessor(task)
+
+    with patch.object(job, "update_job_state") as mock_update_job_state:
+        processor.start()
+
+    task.refresh_from_db()
+    assert task.status == AnalysisTask.Status.CANCELED
+    assert task.started_at is not None
+    assert task.ended_at is not None
+    mock_update_job_state.assert_called_once()
+
+
+@pytest.mark.django_db
+def test_start_with_canceled_job_status_canceled():
+    user = UserFactory.create()
+    job = ExtractionJobFactory.create(owner=user, status=AnalysisJob.Status.CANCELED)
+    task = ExtractionTaskFactory.create(job=job, status=AnalysisTask.Status.PENDING)
+
+    processor = AnalysisTaskProcessor(task)
+
+    with patch.object(job, "update_job_state") as mock_update_job_state:
+        processor.start()
+
+    task.refresh_from_db()
+    assert task.status == AnalysisTask.Status.CANCELED
+    assert task.started_at is not None
+    assert task.ended_at is not None
+    mock_update_job_state.assert_called_once()
+
+
+@pytest.mark.django_db
+def test_start_with_canceled_task_status():
+    user = UserFactory.create()
+    job = ExtractionJobFactory.create(owner=user, status=AnalysisJob.Status.PENDING)
+    task = ExtractionTaskFactory.create(job=job, status=AnalysisTask.Status.CANCELED)
+
+    processor = AnalysisTaskProcessor(task)
+
+    with patch.object(job, "update_job_state") as mock_update_job_state:
+        processor.start()
+
+    task.refresh_from_db()
+    assert task.status == AnalysisTask.Status.CANCELED
+    assert task.started_at is not None
+    assert task.ended_at is not None
+    mock_update_job_state.assert_called_once()
+
+
+@pytest.mark.django_db
+@time_machine.travel("2025-01-15 14:30:00+00:00")
+def test_start_job_transition_from_pending_to_in_progress():
+    user = UserFactory.create()
+    job = ExtractionJobFactory.create(owner=user, status=AnalysisJob.Status.PENDING)
+    task = ExtractionTaskFactory.create(job=job, status=AnalysisTask.Status.PENDING)
+
+    processor = AnalysisTaskProcessor(task)
+
+    with (
+        patch.object(processor, "process_task") as mock_process_task,
+        patch.object(job, "update_job_state") as mock_update_job_state,
+    ):
+        processor.start()
+
+    job.refresh_from_db()
+    task.refresh_from_db()
+
+    expected_time = timezone.now()
+    assert job.status == AnalysisJob.Status.IN_PROGRESS
+    assert job.started_at is not None
+    assert abs((job.started_at - expected_time).total_seconds()) < 1
```

Comment:
I don't fully get that. Shouldn't the time be exactly like the time we travelled to with time_machine?